File: sorts/selection_sort.py
def selection_sort(input):
    for i in range(len(input)):
        min = i
        for j in range(i + 1, len(input)):
            if input[min] > input[j]:
                min = j
        input[i], input[min] = input[min], input[i]
    return input


print(selection_sort([3, 4, 1, 2, 3, 10, 9, 7]))

File: sorts/javascript/selection-sort.js
const selectionSort = (array) => {
    let result = [...array];
    for (let i = 0; i < result.length; i++) {
        let min = i;
        for (let j = i; j < result.length; j++) {
            if (result[min] > result[j]) {
                min = j;
            }
        }
        let swap = result[min];
        result[min] = result[i];
        result[i] = swap;
    }
    return result;
};

let result = selectionSort([5, 7, 8, 1, 10, 12, 14]);
console.log(...result);
